pkg/actions/toolkit/enable.go

```diff
@@ -0,0 +1,123 @@
+package toolkit
+
+import (
+	"context"
+
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/flux"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
+)
+
+const allNamespaces = ""
+
+var defaultFluxComponents = []string{"helm-controller", "kustomize-controller", "notification-controller", "source-controller"}
+
+//go:generate counterfeiter -o fakes/fake_flux_client.go . FluxClient
+type FluxClient interface {
+	PreFlight() error
+	Bootstrap(opts *api.Toolkit) error
+}
+
+type Manager struct {
+	Opts         *api.Toolkit
+	K8sClientSet kubeclient.Interface
+	FluxClient   FluxClient
+}
+
+func New(k8sClientSet kubeclient.Interface, opts *api.Git) (*Manager, error) {
+	if opts.Toolkit == nil {
+		return nil, errors.New("expected git.toolkit in cluster configuration but found nil")
+	}
+
+	m := &Manager{
+		Opts:         opts.Toolkit,
+		K8sClientSet: k8sClientSet,
+		FluxClient:   flux.NewClient(opts.Toolkit.AuthTokenPath, opts.Toolkit.GitProvider),
+	}
+
+	return m, nil
+}
+
+func (m *Manager) Run() error {
+	// TODO for now if we discover V1 components we just abort
+	// in future we may want to handle some migration magic
+	logger.Info("ensuring v1 repo components not installed")
+	v1Namespace, err := m.checkV1()
+	if err != nil {
+		return errors.Wrap(err, "checking for flux v1 components")
+	}
+
+	if v1Namespace != "" {
+		logger.Warning("flux v1 components already installed in namespace %q. auto-migration not yet supported by eksctl. skipping installation", v1Namespace)
+		return nil
+	}
```

Comment:
could `checkV1` just return an error, wrap the logic for checking namespace value inside the check?

Reply:
oh lol i actually had it like that before. have no memory of how it got like this

Reply:
oh now i remember. we need to abort if something is found, but finding something is not an error so i don't want to just return the error (the logging is nicer), so it was this or use a bool.